### ISSUE TITLE
Darktable errors and crash with OpenCL

### ISSUE THREAD

**Issue by kofa73** [darktable-org/darktable#17960]
### Describe the bug

Darktable crashes with filmic_rgb autotune, if OpenCL is active.

Alternatively (possibly after such a crash), darktable disables OpenCL due to error; see attached log.
[opencl-error.txt](https://github.com/user-attachments/files/18029598/opencl-error.txt)
[darktable_bt_K54SY2.txt](https://github.com/user-attachments/files/18029609/darktable_bt_K54SY2.txt)

According to git bisect:
```text
kofa@eagle:~/darktable$ git bisect bad
031fbb478175868f77fac7bd6c32b0ad7266c18d is the first bad commit
commit 031fbb478175868f77fac7bd6c32b0ad7266c18d (HEAD)
Author: Hanno Schwalm <hanno@schwalm-bremen.de>
Date:   Wed Dec 4 19:03:21 2024 +0100

    Some OpenCL fixes
    
    1. Two functions were purely internal and thus are not exposed any more and got a leading underscore
         _opencl_set_kernel_arg() and _opencl_get_mem_context_id()
    2. _opencl_set_kernel_arg() returns an error condition so it generate them instead of magics.
       We also should test vs CL_SUCCESS instead of assuming this is zero
    3. dt_opencl_events_flush() returns an error code so it should generate them and we should test
       results vs CL_SUCCESS.
    
    Some codestyle fixes and readability

 src/common/opencl.c        | 63 ++++++++++++++++++++++++++++++++-------------------------------
 src/common/opencl.h        |  9 ---------
 src/develop/pixelpipe_hb.c |  2 +-
 3 files changed, 33 insertions(+), 41 deletions(-)
```

### Steps to reproduce

Open any raw
Activate filmic
Toggle the auto-tune

### Expected behavior

Should not crash/hang

### Commit

031fbb47817

### Where did you obtain darktable from?

self compiled

### darktable version

031fbb47817

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 24.10

### Describe your system?

Proprietary driver because of GTX 1060 not supported by open-source 560 series.

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

nvidia-driver-550-server 550.127.08-0ubuntu0.24.10.1

**Comment by jenshannoschwalm** [comment]
Got it. you disabled Events?

**Comment by kofa73** [comment]
> Got it. you disabled Events?

Yep. Never had issues, thought it could give a minuscule performance enhancement. Now it bit me. There's a word for that in Hungarian: "önsorsrontó" (someone making oneself miserable / making this/her life more difficult).

I can confirm your PR fixed the issue.

**Comment by jenshannoschwalm** [comment]
The good point is, in the before-the-bad-commit code we could run into CPU fallbacks I never understood that are now fixed.

**Comment by jenshannoschwalm** [comment]
Btw I don't know of a one-word German expression for that.

**Comment by kofa73** [comment]
> Btw I don't know of a one-word German expression for that.

And we have no one-word expression for "wohlstandsverwahrlost", so let's call it even. :-D
